<launch>

    <node name="pcloud_stich" pkg="vivavis_vision" type="vivavis_pcloud_stich" respawn="false"
        output="screen">
        <remap from="/in_pointcloud" to="/camera/depth_registered/points" />
        <!-- <remap from="/stiched_pointcloud" to="/camera/depth_registered/points"/> -->

        <param name="voxel_size_stiching" value="0.1" />
        <param name="voxel_size_input_cloud" value="0.1" />
        <param name="max_cam_depth" value="3.0" />
        <param name="output_reference_frame" value="world" />
    </node>

    <!-- <node name="vivavis_vision" pkg="vivavis_vision" type="vivavis_vision_node" respawn="false"
    launch-prefix="gdb -ex run - -args" 
        output="screen"> -->
    <node name="vivavis_vision" pkg="vivavis_vision" type="vivavis_vision_node" respawn="false"
        output="screen">
        <param name="optical_frame" type="string" value="camera_color_optical_frame" />
        <param name="fixed_frame" type="string" value="world" />

        <param name="far_clip" type="double" value="4.0" />
        <param name="near_clip" type="double" value="0.01" />

        <!-- <param name="orig_cld_voxel_size" value="0.05" /> -->
        <param name="orig_cld_voxel_size" value="0.05" /> <!-- 0.1 -->

        <param name="object_cluster_distance" type="double" value="3" />
        <param name="min_object_cluster_size" type="int" value="1" />

        <remap from="in_cloud" to="/stiched_pointcloud" />
        <!-- <remap from="in_cloud" to="/camera/depth_registered/points"/> -->
        <!-- <remap from="in_cloud" to="/slam_registration/sparse_pointcloud" /> -->
        <!-- <remap from="walls_cloud" to="/planes"/> -->


        <remap from="debug_pose" to="/vivavis/ellipsoid/debug_pose" />
        <!-- private handler-->
        <remap from="vivavis_vision/walls_cloud" to="/vivavis/filter_cloud" />
        <remap from="vivavis_vision/out_cloud_array" to="/vivavis/cloud_array" />
        <remap from="vivavis_vision/ellipsoid" to="/vivavis/ellipsoid_data" />
        <remap from="vivavis_vision/ellipsoid_cloud" to="/vivavis/ellipsoid_cloud" />

    </node>

    <node name="rviz_vivavis" pkg="rviz" type="rviz"
        args="-d $(find vivavis_vision)/rviz/vivavis_vision.rviz" respawn="false" />


</launch>